### PR TITLE
Update SingleTableRuleTest.java

### PR DESCRIPTION
**Pull request by zzs52**
Add test cases for SingleTableRule #12215.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

**Comment by zzs52** [comment]
Hi,CI prompts me to have 1 Checkstyle violation.But I didn't check it out.Can you tell me where is the problem I would be grateful?

**Comment by strongduanmu** [comment]
> Can

@zzs52 src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java:[154] (coding) VariableDeclarationUsageDistance: Distance between variable 'singleTableRule' declaration and its first usage is 4, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java [hunk 1]
```diff
@@ -87,7 +91,7 @@ public void assertGetRuleType() {
                 Collections.emptyMap(), Collections.emptyList(), new ConfigurationProperties(new Properties()));
         assertThat(singleTableRule.getType(), is(SingleTableRule.class.getSimpleName()));
     }
-
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java:90]: @zzs52 Please keep indents consistent with the previous one.

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java [hunk 2]
```diff
@@ -133,4 +137,138 @@ public void assertFindSingleTableDataNodeWithUpperCase() {
         assertThat(actual.get().getDataSourceName(), is("ds_0"));
         assertThat(actual.get().getTableName(), is("employee"));
     }
+
+    @Test
+    public void assertIsSingleTablesInSameDataSource() {
+        DataNodeContainedRule dataNodeContainedRule = mock(DataNodeContainedRule.class);
+        SingleTableRule singleTableRule = new SingleTableRule(new SingleTableRuleConfiguration(), mock(DatabaseType.class), dataSourceMap,
+                Collections.singletonList(dataNodeContainedRule), new ConfigurationProperties(new Properties()));
+        Collection<String> singleTableNames = new ArrayList<>();
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java:146]: LinkedList may be better.

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java [hunk 3]
```diff
@@ -133,4 +137,138 @@ public void assertFindSingleTableDataNodeWithUpperCase() {
...
+
+    @Test
+    public void assertIsAllTablesInSameDataSource() {
+        DataNodeContainedRule dataNodeContainedRule = mock(DataNodeContainedRule.class);
+        SingleTableRule singleTableRule = new SingleTableRule(new SingleTableRuleConfiguration(), mock(DatabaseType.class), dataSourceMap,
+                Collections.singletonList(dataNodeContainedRule), new ConfigurationProperties(new Properties()));
+        Collection<String> singleTableNames = new ArrayList<>();
+        singleTableNames.add("employee");
+        RouteMapper dataSourceMapper = new RouteMapper("ds_0", null);
+        Collection<RouteMapper> tableMappers = new ArrayList<>();
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java:159]: Same problem.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java (hunk 3 above)
Comment (zzs52) [reply shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java]: > Same problem.

Thank you so much.

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java [hunk 4]
```diff
@@ -32,44 +34,46 @@
...
 import static org.junit.Assert.assertTrue;
 import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 public final class SingleTableRuleTest {
-    
+
     private static final String TABLE_TYPE = "TABLE";
-
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/SingleTableRuleTest.java:51]: @zzs52 Please keep indents consistent with the previous one.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
@zzs52 Good job, please go ahead.

**Comment by zzs52** [comment]
> @zzs52 Good job, please go ahead.

Thanks for the encouragement, I will.

**Pull request merged**